backend/devices/bitbox02/keystore_simulator_test.go

```diff
@@ -412,12 +419,77 @@ func makeTx(t *testing.T, device *Device, recipient *maketx.OutputInfo) *btc.Pro
 		GetPrevTx: func(chainhash.Hash) (*wire.MsgTx, error) {
 			return prevTx, nil
 		},
+		GetAddress: func(scriptHashHex blockchain.ScriptHashHex) (*addresses.AccountAddress, error) {
+			for _, address := range addrsInDifferentAccount {
+				if address.PubkeyScriptHashHex() == scriptHashHex {
+					return address, nil
+				}
+			}
+			return nil, nil
+		},
 		Signatures: make([]*types.Signature, len(txProposal.Transaction.TxIn)),
 		FormatUnit: coinpkg.BtcUnitDefault,
 	}
 
 }
 
+func TestSimulatorSignBTCTransactionSendSelfDifferentAccount(t *testing.T) {
+	testInitializedSimulators(t, func(t *testing.T, device *Device, stdOut *bytes.Buffer) {
+		t.Helper()
+
+		cfg := makeConfig(t, device, signing.ScriptTypeP2WPKH, mustKeypath("m/84'/0'/1'"))
+		selfAddress := addresses.NewAccountAddress(
+			cfg,
+			signing.NewEmptyRelativeKeypath().Child(0, false).Child(0, false),
+			network,
+			log)
+		proposedTransaction := makeTx(t, device, maketx.NewOutputInfo(selfAddress.PubkeyScript()))
+
+		require.NoError(t, device.Keystore().SignTransaction(proposedTransaction))
+		require.NoError(t, proposedTransaction.Finalize())
+		require.NoError(
+			t,
+			btc.TxValidityCheck(
+				proposedTransaction.TXProposal.Transaction,
+				proposedTransaction.TXProposal.PreviousOutputs,
+				proposedTransaction.TXProposal.SigHashes()))
+
+		// Send-to-self message from different accounts is only available on v9.22 and later.
+		if device.Version().AtLeast(semver.NewSemVer(9, 22, 0)) {
+			require.Contains(t, stdOut.String(), "ADDRESS: This BitBox (account #2): bc1qeunlpt8umlyyv0mpzpe3uauqlezsh39ctwg4e5")
+		}
```

Comment:
I'd still add an `else { ... }` and include the expected output for v9.20-v9.21, for clarity.

Reply:
Done